### PR TITLE
Adding IO component

### PR DESCRIPTION
**Pull request by nkoenig**
Signed-off-by: Nate Koenig <nate@openrobotics.org>

Testing here: https://build.osrfoundation.org/job/gz-common5-debbuilder/806/

### CODE REVIEW

File: ubuntu/debian/control [hunk 1]
```diff
@@ -234,6 +235,42 @@ Description: Collection of useful code used by robotics apps - Graphics dev file
...
 
+Package: libgz-common5-io
+Architecture: any
+Section: libs
+Pre-Depends: ${misc:Pre-Depends}
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Breaks: libignition-common5-io (<< 4.999.999+nightly+git20220630+1r4416bee895d08f2907d718fea34bbd09df6d0c88-1)
+Replaces: libignition-common5-io (<< 4.999.999+nightly+git20220630+1r4416bee895d08f2907d718fea34bbd09df6d0c88-1)
+Multi-Arch: same
+Description: Collection of useful code used by robotics apps - Graphics libraries
```
Comment (scpeters) [inline ubuntu/debian/control]:
```suggestion
Description: Collection of useful code used by robotics apps - I/O libraries
```

File: ubuntu/debian/control [hunk 2]
```diff
@@ -234,6 +235,42 @@ Description: Collection of useful code used by robotics apps - Graphics dev file
...
+Depends: libgz-cmake3-dev,
+         libgz-common5-core-dev (= ${binary:Version}),
+         libgz-math7-dev,
+         libgz-utils2-dev,
+         libgz-common5-io (= ${binary:Version}),
+         ${misc:Depends}
+Breaks: libignition-common5-io-dev (<< 4.999.999+nightly+git20220630+1r4416bee895d08f2907d718fea34bbd09df6d0c88-1)
+Replaces: libignition-common5-io-dev (<< 4.999.999+nightly+git20220630+1r4416bee895d08f2907d718fea34bbd09df6d0c88-1)
+Multi-Arch: same
+Description: Collection of useful code used by robotics apps - Graphics dev files
```
Comment (scpeters) [inline ubuntu/debian/control]:
```suggestion
Description: Collection of useful code used by robotics apps - I/O dev files
```

**Comment by scpeters** [comment]
I think it also needs `.install` files like the following:

* https://github.com/gazebo-release/gz-common5-release/blob/main/ubuntu/debian/libgz-common-av-dev.install
* https://github.com/gazebo-release/gz-common5-release/blob/main/ubuntu/debian/libgz-common-av.install

the naming and sym-linking is a little tricky. The `ubuntu/debian` files are not versioned, but the sym-links in `focal/debian` and `jammy/debian` have version numbers

**Comment by nkoenig** [comment]
Thanks, forgot to add in those files. [9e6a826](https://github.com/gazebo-release/gz-common5-release/pull/22/commits/9e6a826ca61bf902426cf6bdfe0ef9ce252e9c48)

File: ubuntu/debian/control (hunk 1 above)
Comment (nkoenig) [reply ubuntu/debian/control]: Done

File: ubuntu/debian/control (hunk 2 above)
Comment (nkoenig) [reply ubuntu/debian/control]: Done.

**Comment by nkoenig** [comment]
New test: https://build.osrfoundation.org/job/gz-common5-debbuilder/807/

**Comment by nkoenig** [comment]
Arg, I think I forgot the symlinks

**Comment by nkoenig** [comment]
Looks to be working now: https://build.osrfoundation.org/job/gz-common5-debbuilder/808/

**Review by scpeters — APPROVED** [review]

**Pull request merged**